openmls/src/group/core_group/new_from_external_init.rs

```diff
@@ -117,14 +125,30 @@ impl CoreGroup {
         );
         let message_secrets_store = MessageSecretsStore::new_with_secret(0, message_secrets);
 
+        let interim_transcript_hash = {
+            if group_info.group_context().epoch() == GroupEpoch::from(0) {
```

Comment:
I'm not sure this can ever happen.
Joining a group by external commit implies that there is a group already. The commit that is generated for this join moves the epoch forward. I therefore don't see how the epoch can be 0 here.
Is this actually used? Shouldn't we throw an error instead (if the above is correct)?

Reply:
Without this condition, tests fail with a `InvalidCommit(ConfirmationTagMismatch)`. But, to be honest, I am not entirely sure the draft tells us what to do in this case.

In the failing tests, alice creates a group first (with no other members).

According to draft-16, the group should have ...

> Initialize a one-member group with the following initial values:
> [...]
> epoch: 0
> [...]
> Confirmed transcript hash: The zero-length octet string
> Interim transcript hash: The zero-length octet string

(In OpenMLS, the `interim_transcript_hash` is set to `b"\x00" * 32` and `confirmed_transcript_hash` is non-zero. I assume the `confirmed_transcript_hash` is correct.)

Then, alice exports a group state and bob joins by an external commit. Here, the epoch is 0 (and triggers the above condition).

Now, I am not sure, how the `interim_transcript_hash` should be calculated really. From my understanding of draft-16, I would expect the hash of a zero-length string (empty interim || empty confirmation). However, debugging the "old code" (with `PublicGroupState`) it seemed to me that `b""` was used instead. Given that all tests pass having this condition, I assumed it to be fine.

Could you shed some light on this`

Reply:
I've created https://github.com/openmls/openmls/issues/1129.

Reply:
Can you link the issue here in the code as well?